### PR TITLE
Update CameraTraps Batch API to utilize Megadetector V5

### PR DESCRIPTION
**Pull request by yuenmichelle1**
**Changes**
- Pulled changes from base (microsoft/CameraTraps. NOTE that pulled before they archived CameraTraps in favor of PyTorch Wildlife. Megadetector and CameraTraps are still being supported via agentmorris. https://github.com/agentmorris/MegaDetector)
- Update Dockerfile to base off of Pytorch (MD v5 utilizes YOLOV5 [for object detection and image classification] and Pytorch)
- create a pytorch_detector.py based on detection/pytorch_detector.py and https://github.com/agentmorris/MegaDetector/blob/main/detection/pytorch_detector.py
- Update relevant README to stop referencing TensorFlow and start referencing PyTorch

**Side Note**
Whats nice is that these changes once deployed, should propagate down to hamlet so we really don't need to make any code changes in hamlet or subject-assistant. Meaning once this is deployed, Hamlet's celery worker when calling out to camera-traps api should automatically use MDv5.

### CODE REVIEW

**Comment by yuenmichelle1** [comment]
@zwolf  Feel free to do a full review of PR if you want (most of the file changes are pulled from microsoft cameratraps main repo and unrelated to relevant changes). What I really need a review on is the GH action that replaces the jenkinsfile. 
GH Action
https://github.com/zooniverse/CameraTraps/pull/14/files#diff-53c24665441c6f14455ff3980322a41913e7c3bc35572986213fe21b9167734d

JenkinsFile
https://github.com/zooniverse/CameraTraps/pull/14/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8

If you want to see relevant changes, all relevant changes should live within `api/batch_processing`

File: .github/workflows/deploy_app.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      commit_id: ${{ github.sha }}
+      file: ./api/batch_processing/api_core/Dockerfile
+      latest: true
+
+  deploy_app:
+    runs-on: ubuntu-latest
+    needs: [build_and_push_image]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3.5.2
```
Comment (zwolf) [inline .github/workflows/deploy_app.yml]: I updated to this version in the shared workflows, thanks for the reminder.
```suggestion
      uses: actions/checkout@v4.1.1
```

File: .github/workflows/deploy_app.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v2
```
Comment (zwolf) [inline .github/workflows/deploy_app.yml]:
```suggestion
      uses: docker/login-action@v3
```

**Review by zwolf — CHANGES_REQUESTED** [review]
I agree: the build/push process can use the shared workflow, but the deploy template location doesn't match the assumption in the shared one and so it needs to be separate. Also, couple of those action versions can be updated.

One thing this PR does need is for the kubernetes template to reflect the location of the image you're pushing. I can't add a suggestion to an unchanged file but this line:

https://github.com/zooniverse/CameraTraps/blob/b3b6b62916970797587f44987d030566c321f223/api/batch_processing/api_core/kubernetes/deployment.tmpl#L19

needs to be:
`image: ghcr.io/zooniverse/camera-traps-api:__IMAGE_TAG__`

Finally, just a suggestion: for clarity's sake, maybe you should delete the [other stale deployment branches](https://github.com/zooniverse/CameraTraps/branches/stale) if they're not being used.

**Pull request merged**